### PR TITLE
Fixed small bugs in decodemd.cpp, gcrecord.h, gc.cpp, Strike/util.cpp…

### PR DESCRIPTION
**Pull request by wtgodbe**
…, and strike.cpp, as outlined in this article: http://www.viva64.com/en/b/0310/#ID0E5BJM

### CODE REVIEW

File: src/gc/gcrecord.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ Module Name:
 // We pack the dynamic tuning for deciding which gen to condemn in a DWORD.
 // We assume that 2 bits are enough to represent the generation. 
 #define bits_generation 2
-#define generation_mask (~(~0 << bits_generation))
+#define generation_mask (~(~((unsigned int)0) << bits_generation))
```
Comment (janvorli) [inline src/gc/gcrecord.h]: A fix for this (using "u" suffix instead of the cast) was already merged in as #1650 a day ago

File: src/pal/inc/pal.h [hunk 2]
```diff
@@ -3874,9 +3874,9 @@ PALIMPORT
 HANDLE
 PALAPI
 HeapCreate(
-	       IN DWORD flOptions,
-	       IN SIZE_T dwInitialSize,
-	       IN SIZE_T dwMaximumSize);
+         IN DWORD flOptions,
```
Comment (janvorli) [inline src/pal/inc/pal.h]: Please remove these unnecessary formatting changes.

File: src/vm/decodemd.cpp [hunk 3]
```diff
@@ -73,7 +73,7 @@ const BYTE decoded_10[2] = {10, END_DECODED };
 #define DOING_BITS (MAX_HEADER+1)
 #define DECODING_BITS(partial, got, header) (DOING_BITS+(partial<<8)+(got<<16)+(header<<24))
 #define DECODING_ERROR ((unsigned) -1)
-#define MASK(len) (~((~0)<<len))
+#define MASK(len) (~((~((unsigned int)0))<<len))
```
Comment (janvorli) [inline src/vm/decodemd.cpp]: To make it the same as the already checked in change for similar thing, could you please replace the cast by adding "u" suffix to the 0?

File: src/gc/gc.cpp [hunk 4]
```diff
@@ -22544,7 +22544,7 @@ void gc_heap::make_free_lists (int condemned_gen_number)
 
 void gc_heap::make_free_list_in_brick (BYTE* tree, make_free_args* args)
 {
-    assert ((tree >= 0));
+    assert ((tree > 0));
```
Comment (janvorli) [inline src/gc/gc.cpp]: Since you are touching this, it seem it would be nicer to check for inequality with NULL instead.

**Comment by sergiy-k** [comment]
LGTM

**Comment by janvorli** [comment]
@wtgodbe There seems to be some merge problem with the commit. It shows some of my changes and also Mike's debugging changes that were already merged in some time ago.
Maybe you'd need to rebase your changes against the up to date master.

**Comment by wtgodbe** [comment]
Just did what you suggested, hopefully that’s fixed the problem.

**Comment by janvorli** [comment]
@wtgodbe Thanks, it looks ok now.
